core/src/main/scala/com/databricks/labs/remorph/parsers/snowflake/SnowflakeAstBuilder.scala

```diff
@@ -67,10 +67,16 @@ class SnowflakeAstBuilder(override val vc: SnowflakeVisitorCoordinator)
 
   // TODO: Sort out where to visitSubquery
   override def visitQueryStatement(ctx: QueryStatementContext): ir.LogicalPlan = {
+    errorCheck(ctx).getOrElse {
+      val query = ctx.queryExpression().accept(this)
+      Option(ctx.withExpression()).foldRight(query)(buildCTE)
+    }
+  }
+
+  override def visitQueryExpression(ctx: QueryExpressionContext): ir.LogicalPlan = {
     errorCheck(ctx).getOrElse {
       val select = ctx.selectStatement().accept(vc.relationBuilder)
-      val queryExpression = ctx.setOperators().asScala.foldLeft(select)(buildSetOperator)
-      Option(ctx.withExpression()).foldRight(queryExpression)(buildCTE)
```

Comment:
Does this violate the parameter list stuff, or no?

https://github.com/databricks/scala-style-guide?tab=readme-ov-file#multiple-parameter-lists

Reply:
That's a good question. I interpret that guideline to be about declaring functions that use multiple parameter lists, which the `case class` is effectively doing.

In this example we're using them on a method declared by the standard library, which is not what I think the guideline is aimed at. (Such usage is unavoidable due to the way Scala libraries are often designed, especially if they take a function/closure as a parameter.)